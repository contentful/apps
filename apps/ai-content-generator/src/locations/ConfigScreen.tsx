import ConfigPage from '@components/config/config-page/ConfigPage';
import { ProfileFields } from '@components/config/configText';

export type ProfileType = {
  [K in ProfileFields]?: string;
};
interface AppInstallationParameters {
  // Model is optional since V1 of the config page didn't save model as a param if you had the default selected.
  model?: string;
  key: string;
  profile: string;
  // Optional since we are accessing properties on this object
  brandProfile?: ProfileType;
}

const ConfigScreen = () => {
  return <ConfigPage />;
};

export default ConfigScreen;
export type { AppInstallationParameters };
