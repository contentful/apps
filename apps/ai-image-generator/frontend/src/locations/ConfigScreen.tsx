import { useCallback, useState, useEffect } from 'react';
import { ConfigAppSDK } from '@contentful/app-sdk';
import { useSDK } from '@contentful/react-apps-toolkit';
import ConfigPage from 'components/config/ConfigPage/ConfigPage';
import { configPageErrorMessages } from 'constants/errorMessages';
import { AppInstallationParameters, InstallErrors } from 'types/configPage';

const ConfigScreen = () => {
  const [parameters, setParameters] = useState<AppInstallationParameters>({});
  const [appIsInstalled, setAppIsInstalled] = useState<boolean>(false);
  const sdk = useSDK<ConfigAppSDK>();

  useEffect(() => {
    const getAppActions = async () => {
      const appActions = await sdk.cma.appAction.getMany({
        appDefinitionId: sdk.ids.app,
        limit: 100,
      });
      console.log('appActions', appActions);
    };

    getAppActions();
  }, [sdk.cma.appAction, sdk.ids.app]);

  const handleInstall = useCallback(async () => {
    const currentState = await sdk.app.getCurrentState();

    const handleError = (errorType: InstallErrors) => {
      sdk.notifier.error(configPageErrorMessages[errorType]);
    };

    if (!parameters.apiKey) {
      handleError('apiKeyEmpty');
      return false;
    }

    setAppIsInstalled(true);

    return {
      parameters,
      targetState: currentState,
    };
  }, [parameters, sdk]);

  useEffect(() => {
    sdk.app.onConfigure(() => handleInstall());
  }, [sdk, handleInstall]);

  useEffect(() => {
    (async () => {
      const currentParameters: AppInstallationParameters | null = await sdk.app.getParameters();

      if (currentParameters) {
        setParameters(currentParameters);
      }

      sdk.app.setReady();
    })();
  }, [sdk]);

  useEffect(() => {
    async function checkAppIsInstalled() {
      const installed = await sdk.app.isInstalled();
      if (installed) {
        setAppIsInstalled(true);
      }
    }
    checkAppIsInstalled();
  }, [sdk]);

  const handleConfig = (updatedParams: AppInstallationParameters) => {
    setParameters((prevParameters) => ({
      ...prevParameters,
      ...updatedParams,
    }));
  };

  return (
    <ConfigPage
      parameters={parameters}
      handleConfig={handleConfig}
      appIsInstalled={appIsInstalled}
    />
  );
};

export default ConfigScreen;
