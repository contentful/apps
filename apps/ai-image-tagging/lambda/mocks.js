'use strict';

const fetch = jest.fn().mockResolvedValue({
  status: 200,
  arrayBuffer: jest.fn().mockResolvedValue('SOME_ARR_BUFF')
});

const documentClient = {
  update: jest.fn().mockReturnValue({
    promise: jest.fn().mockResolvedValue({
      Attributes: {
        reqs: 7
      }
    })
  })
};

const rekog = {
  detectLabels: jest.fn().mockReturnValue({
    promise: jest.fn().mockResolvedValue({
      Labels: [
        { Name: 'cat' },
        { Name: 'yolo' }
      ]
    })
  })
};

module.exports = {
  fetch,
  documentClient,
  rekog
};
