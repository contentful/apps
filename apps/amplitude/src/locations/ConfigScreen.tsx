import { useCallback, useState, useEffect, ChangeEvent } from 'react';
import { ConfigAppSDK } from '@contentful/app-sdk';
import {
  Box,
  Flex,
  FormControl,
  Heading,
  HelpText,
  Paragraph,
  Stack,
  TextInput,
  TextLink,
} from '@contentful/f36-components';
import { useSDK } from '@contentful/react-apps-toolkit';
import { ExternalLinkIcon } from '@contentful/f36-icons';

export interface AppInstallationParameters {
  amplitudeApiKey?: string;
  amplitudeSecretKey?: string;
  amplitudeProjectId?: string;
}

const ConfigScreen = () => {
  console.log('⚙️ Amplitude Config: Initializing configuration screen');
  
  const sdk = useSDK<ConfigAppSDK>();
  const [parameters, setParameters] = useState<AppInstallationParameters>({});
  const [amplitudeApiKey, setAmplitudeApiKey] = useState<string>('');
  const [amplitudeSecretKey, setAmplitudeSecretKey] = useState<string>('');
  const [amplitudeProjectId, setAmplitudeProjectId] = useState<string>('');

  console.log('📊 Amplitude Config: Current parameters state:', parameters);

  const validateInputs = useCallback(() => {
    console.log('🔍 Amplitude Config: Validating configuration inputs...');
    
    const isApiKeyValid = amplitudeApiKey.trim().length > 0;
    const isSecretKeyValid = amplitudeSecretKey.trim().length > 0;
    const isProjectIdValid = amplitudeProjectId.trim().length > 0;
    
    console.log('✅ Amplitude Config: API Key valid:', isApiKeyValid);
    console.log('✅ Amplitude Config: Secret Key valid:', isSecretKeyValid);
    console.log('✅ Amplitude Config: Project ID valid:', isProjectIdValid);
    
    return isApiKeyValid && isSecretKeyValid && isProjectIdValid;
  }, [amplitudeApiKey, amplitudeSecretKey, amplitudeProjectId]);

  const onConfigure = useCallback(async () => {
    console.log('💾 Amplitude Config: Starting configuration save process...');
    
    const currentState = await sdk.app.getCurrentState();
    console.log('📋 Amplitude Config: Current app state retrieved:', currentState);

    if (!validateInputs()) {
      console.error('❌ Amplitude Config: Validation failed - missing required fields');
      sdk.notifier.error('Please provide all required Amplitude credentials.');
      return false;
    }

    const parametersToSave = {
      ...parameters,
      amplitudeApiKey,
      amplitudeSecretKey,
      amplitudeProjectId,
    };

    console.log('💾 Amplitude Config: Saving parameters:', {
      ...parametersToSave,
      amplitudeSecretKey: '[REDACTED]' // Don't log secret in production
    });

    setParameters(parametersToSave);

    console.log('✅ Amplitude Config: Configuration save completed successfully');
    
    return {
      parameters: parametersToSave,
      targetState: currentState,
    };
  }, [parameters, sdk, amplitudeApiKey, amplitudeSecretKey, amplitudeProjectId, validateInputs]);

  useEffect(() => {
    console.log('🔗 Amplitude Config: Setting up configuration callback');
    sdk.app.onConfigure(() => onConfigure());
  }, [sdk, onConfigure]);

  useEffect(() => {
    console.log('🚀 Amplitude Config: Loading existing configuration parameters...');
    
    (async () => {
      try {
        const currentParameters = await sdk.app.getParameters();
        console.log('📥 Amplitude Config: Retrieved existing parameters:', currentParameters ? 'Found' : 'None');

        if (currentParameters) {
          setParameters(currentParameters);
          setAmplitudeApiKey(currentParameters.amplitudeApiKey || '');
          setAmplitudeSecretKey(currentParameters.amplitudeSecretKey || '');
          setAmplitudeProjectId(currentParameters.amplitudeProjectId || '');
          console.log('✅ Amplitude Config: Parameters loaded successfully');
        }

        sdk.app.setReady();
        console.log('🎯 Amplitude Config: App marked as ready for user interaction');
      } catch (error) {
        console.error('❌ Amplitude Config: Error loading parameters:', error);
      }
    })();
  }, [sdk]);

  const handleApiKeyChange = (e: ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    console.log('📝 Amplitude Config: API Key updated, length:', newValue.length);
    setAmplitudeApiKey(newValue);
  };

  const handleSecretKeyChange = (e: ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    console.log('📝 Amplitude Config: Secret Key updated, length:', newValue.length);
    setAmplitudeSecretKey(newValue);
  };

  const handleProjectIdChange = (e: ChangeEvent<HTMLInputElement>) => {
    const newValue = e.target.value;
    console.log('📝 Amplitude Config: Project ID updated:', newValue);
    setAmplitudeProjectId(newValue);
  };

  console.log('🎨 Amplitude Config: Rendering configuration screen');

  return (
    <Box padding="spacingL">
      <Box marginBottom="spacingL">
        <Heading>Set Up Amplitude Analytics</Heading>
        <Paragraph>
          Connect your Amplitude analytics to track and analyze user behavior in your Contentful entries.
          This integration allows you to send events and monitor content performance.
        </Paragraph>
      </Box>
      
      <Stack spacing="spacingL" flexDirection="column">
        <FormControl id="apiKey" isRequired={true}>
          <FormControl.Label aria-label="apiKey" htmlFor="apiKey">
            Amplitude API Key
          </FormControl.Label>
          <TextInput
            testId="apiKey"
            spellCheck={false}
            name="apiKey"
            placeholder="Enter your Amplitude API Key"
            value={amplitudeApiKey}
            onChange={handleApiKeyChange}
          />
          <HelpText>
            Find your API Key in the{' '}
            <TextLink
              icon={<ExternalLinkIcon />}
              alignIcon="end"
              href="https://amplitude.com/docs/apis/analytics/find-api-credentials"
              target="_blank"
              rel="noopener noreferrer">
              Amplitude dashboard
            </TextLink>
          </HelpText>
        </FormControl>

        <FormControl id="secretKey" isRequired={true}>
          <FormControl.Label aria-label="secretKey" htmlFor="secretKey">
            Amplitude Secret Key
          </FormControl.Label>
          <TextInput
            testId="secretKey"
            spellCheck={false}
            name="secretKey"
            type="password"
            placeholder="Enter your Amplitude Secret Key"
            value={amplitudeSecretKey}
            onChange={handleSecretKeyChange}
          />
          <HelpText>
            Keep your secret key secure and never share it publicly.
          </HelpText>
        </FormControl>

        <FormControl id="projectId" isRequired={true}>
          <FormControl.Label aria-label="projectId" htmlFor="projectId">
            Amplitude Project ID
          </FormControl.Label>
          <TextInput
            testId="projectId"
            spellCheck={false}
            name="projectId"
            placeholder="Enter your Amplitude Project ID"
            value={amplitudeProjectId}
            onChange={handleProjectIdChange}
          />
          <HelpText>
            The Project ID can be found in your Amplitude project settings.
          </HelpText>
        </FormControl>
      </Stack>
    </Box>
  );
};

export default ConfigScreen;
