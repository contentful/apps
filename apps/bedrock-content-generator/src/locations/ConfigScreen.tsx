import ConfigPage from '@components/config/config-page/ConfigPage';

const ConfigScreen = () => {
  return <ConfigPage />;
};

export default ConfigScreen;
