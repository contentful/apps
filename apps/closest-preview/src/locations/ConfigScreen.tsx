import { useCallback, useState, useEffect } from 'react';
import { ConfigAppSDK } from '@contentful/app-sdk';
import { Heading, Form, Paragraph, Flex, Box, FormControl } from '@contentful/f36-components';
import { useCMA, useSDK } from '@contentful/react-apps-toolkit';
import ContentTypeMultiSelect from '../components/ContentTypeMultiSelect';
import { ContentType } from '../types';

const ConfigScreen = () => {
  const sdk = useSDK<ConfigAppSDK>();
  const cma = useCMA();
  const [selectedContentTypes, setSelectedContentTypes] = useState<ContentType[]>([]);

  const onConfigure = useCallback(async () => {
    const editorInterface = selectedContentTypes.reduce((acc, contentType) => {
      return {
        ...acc,
        [contentType.id]: {
          sidebar: { position: 0 },
        },
      };
    }, {});

    const currentState = await sdk.app.getCurrentState();

    return {
      targetState: {
        ...currentState,
        EditorInterface: editorInterface,
      },
    };
  }, [sdk, selectedContentTypes]);

  useEffect(() => {
    sdk.app.onConfigure(() => onConfigure());
  }, [sdk, onConfigure]);

  useEffect(() => {
    (async () => {
      sdk.app.setReady();
    })();
  }, []);

  return (
    <Flex justifyContent="center" alignItems="flex-start">
      <Box margin="spacing2Xl">
        <Form>
          <Heading as="h2" marginBottom="spacingS">
            Set up Closest Preview
          </Heading>
          <Paragraph marginBottom="spacingXl">
            Closest Preview allows users to quickly navigate to the closest page level element for a
            given entry in order to preview the item.
          </Paragraph>

          <Box marginBottom="spacing2Xl">
            <Heading as="h3" marginBottom="spacingXs">
              Assign content types
            </Heading>
            <Paragraph marginBottom="spacingL">
              Select the content type(s) you want to use with Closest Preview. You can change this
              anytime by navigating to the 'Sidebar' tab in your content model.
            </Paragraph>
            <FormControl id="contentTypes">
              <FormControl.Label>Content types</FormControl.Label>
              <ContentTypeMultiSelect
                selectedContentTypes={selectedContentTypes}
                setSelectedContentTypes={setSelectedContentTypes}
                sdk={sdk}
                cma={cma}
              />
            </FormControl>
          </Box>
        </Form>
      </Box>
    </Flex>
  );
};

export default ConfigScreen;
