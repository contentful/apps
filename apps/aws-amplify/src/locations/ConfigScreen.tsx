import { useCallback, useState, useEffect, ChangeEvent } from 'react';
import { ConfigAppSDK } from '@contentful/app-sdk';
import {
  Box,
  Flex,
  FormControl,
  Heading,
  HelpText,
  Paragraph,
  Stack,
  TextInput,
  TextLink,
} from '@contentful/f36-components';
import { useSDK } from '@contentful/react-apps-toolkit';
import { ExternalLinkIcon } from '@contentful/f36-icons';
import { isValidUrl } from '../lib/isVaildUrl';
import { styles } from '../components/config-screen/ConfigScreen.styles';
import AmplifyIcon from '../components/common/AmplifyIcon';

export interface AppInstallationParameters {
  amplifyWebhookUrl?: string;
}

const ConfigScreen = () => {
  const sdk = useSDK<ConfigAppSDK>();
  const [parameters, setParameters] = useState<AppInstallationParameters>({});
  const [amplifyWebhookUrl, setAmplifyWebhookUrl] = useState<string>('');

  const onConfigure = useCallback(async () => {
    const currentState = await sdk.app.getCurrentState();

    if (!isValidUrl(amplifyWebhookUrl)) {
      sdk.notifier.error('Please provide a valid webhook URL.');
      return false;
    }

    const parametersToSave = {
      ...parameters,
      amplifyWebhookUrl,
    };

    setParameters(parametersToSave);

    return {
      parameters: parametersToSave,
      targetState: currentState,
    };
  }, [parameters, sdk, amplifyWebhookUrl]);

  useEffect(() => {
    sdk.app.onConfigure(() => onConfigure());
  }, [sdk, onConfigure]);

  useEffect(() => {
    (async () => {
      const currentParameters = await sdk.app.getParameters();

      if (currentParameters) {
        setParameters(currentParameters);
        setAmplifyWebhookUrl(currentParameters.amplifyWebhookUrl);
      }

      sdk.app.setReady();
    })();
  }, [sdk]);

  const handleWebhookChange = (e: ChangeEvent<HTMLInputElement>) => {
    setAmplifyWebhookUrl(e.target.value);
  };

  return (
    <>
      <Box className={styles.background} />
      <Box className={styles.body}>
        <Box>
          <Heading>Set Up AWS Amplify</Heading>
          <Paragraph>
            Amplify is a set of purpose-built tools and features that enables frontend web and
            mobile developers to quickly and easily build full-stack applications on AWS.
          </Paragraph>
        </Box>
        <Stack spacing="spacingL" flexDirection="column">
          <Box style={styles.box}>
            <FormControl id="webhookUrl" isRequired={true}>
              <FormControl.Label aria-label="webhookUrl" htmlFor="webhookUrl">
                AWS Amplify Webhook URL
              </FormControl.Label>
              <TextInput
                testId="webhookUrl"
                spellCheck={false}
                name="webhookUrl"
                placeholder={'ex. https://webhooks.amplify.us-east-1.amazonaws.com/...'}
                value={amplifyWebhookUrl}
                onChange={handleWebhookChange}
              />
              <HelpText>
                Follow{' '}
                <TextLink
                  icon={<ExternalLinkIcon />}
                  alignIcon="end"
                  href="https://docs.aws.amazon.com/amplify/latest/userguide/webhooks.html"
                  target="_blank"
                  rel="noopener noreferrer">
                  these detailed instructions
                </TextLink>{' '}
                to create a webhook in the AWS Amplify dashboard.
              </HelpText>
            </FormControl>
          </Box>
        </Stack>
      </Box>
      <Box style={styles.icon}>
        <Flex alignItems="center" justifyContent="center">
          <AmplifyIcon />
        </Flex>
      </Box>
    </>
  );
};

export default ConfigScreen;
