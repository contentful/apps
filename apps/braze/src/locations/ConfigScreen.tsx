import { ConfigAppSDK } from '@contentful/app-sdk';
import {
  Box,
  Flex,
  Form,
  FormControl,
  Heading,
  Paragraph,
  Subheading,
  TextInput,
  TextLink,
  Text,
  Spacing,
} from '@contentful/f36-components';
import { useSDK } from '@contentful/react-apps-toolkit';
import { useCallback, useEffect, useState } from 'react';
import { ExternalLinkIcon } from '@contentful/f36-icons';
import { styles } from './ConfigScreen.styles';
import Splitter from '../components/Splitter';

export interface AppInstallationParameters {
  apiKey: string;
}

export const BRAZE_CONNECTED_CONTENT_DOCUMENTATION =
  'https://braze.com/docs/user_guide/personalization_and_dynamic_content/connected_content';
export const BRAZE_APP_DOCUMENTATION = 'https://www.contentful.com/help/apps/braze-app/';
export const CONTENT_TYPE_DOCUMENTATION =
  'https://www.contentful.com/help/content-types/configure-content-type/';

export async function callToContentful(sdk: ConfigAppSDK, apiKey: string) {
  const url = `https://${sdk.hostnames.delivery}/spaces/${sdk.ids.space}/environments/${sdk.ids.environment}/entries?access_token=${apiKey}`;
  return await fetch(url, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
    },
  });
}

const ConfigScreen = () => {
  const [apiKeyIsValid, setApiKeyIsValid] = useState(true);
  const [parameters, setParameters] = useState<AppInstallationParameters>({
    apiKey: '',
  });
  const sdk = useSDK<ConfigAppSDK>();
  const spaceId = sdk.ids.space;

  async function checkApiKey(apiKey: string) {
    if (!apiKey) {
      setApiKeyIsValid(false);
      return false;
    }

    const response: Response = await callToContentful(sdk, apiKey);

    const isValid = response.ok;
    setApiKeyIsValid(isValid);

    return isValid;
  }

  const onConfigure = useCallback(async () => {
    const currentState = await sdk.app.getCurrentState();

    const isValid = await checkApiKey(parameters.apiKey);

    if (!parameters.apiKey || !isValid) {
      sdk.notifier.error('A valid Contentful API key is required');
      return false;
    }

    return {
      parameters,
      targetState: currentState,
    };
  }, [parameters, sdk]);

  useEffect(() => {
    sdk.app.onConfigure(() => onConfigure());
  }, [sdk, onConfigure]);

  useEffect(() => {
    (async () => {
      const currentParameters: AppInstallationParameters | null = await sdk.app.getParameters();

      if (currentParameters) {
        setParameters(currentParameters);
      }

      sdk.app.setReady();
    })();
  }, [sdk]);

  return (
    <Flex justifyContent="center" alignContent="center">
      <Box className={styles.body} marginTop="spacingL" padding="spacingL">
        <Heading marginBottom="spacingS">Set up Braze</Heading>
        <InformationSection
          url={BRAZE_CONNECTED_CONTENT_DOCUMENTATION}
          linkText="Braze's Connected Content feature">
          The Braze app allows editors to connect content stored in Contentful to Braze campaigns
          through
        </InformationSection>
        <InformationSection
          url={BRAZE_APP_DOCUMENTATION}
          linkText="here"
          marginTop="spacingL"
          dataTestId="braze-app-docs-here-link">
          Learn more about how to connect Contentful with Braze and configure the Braze app
        </InformationSection>
        <Splitter marginTop="spacingL" marginBottom="spacingL" />
        <Heading marginBottom="spacingL">Connected Content configuration</Heading>
        <Subheading className={styles.subheading}>Input the Contentful Delivery API</Subheading>
        <InformationSection
          url={`https://app.contentful.com/spaces/${spaceId}/api/keys`}
          linkText="Manage API Keys">
          Input the Contentful API key that Braze will use to request your content via API at send
          time.
        </InformationSection>
        <Box marginTop="spacingM" marginBottom="spacingM">
          <Form>
            <FormControl.Label>Contentful Delivery API - access token</FormControl.Label>
            <TextInput
              value={parameters.apiKey}
              name="apiKey"
              data-testid="apiKey"
              isInvalid={!apiKeyIsValid}
              placeholder="ex. 0ab1c234DE56f..."
              type="password"
              onChange={(e) => setParameters({ ...parameters, apiKey: e.target.value })}
            />
            {!apiKeyIsValid && (
              <FormControl.ValidationMessage>Invalid API key</FormControl.ValidationMessage>
            )}
          </Form>
        </Box>
        <Subheading className={styles.subheading}>Add Braze to your content types</Subheading>
        <InformationSection
          url={CONTENT_TYPE_DOCUMENTATION}
          linkText="here"
          marginTop="spacing2Xs"
          dataTestId="content-type-docs-here-link">
          Navigate to the content type you would like to use under the Content model tab in the main
          navigation. Select the content type and adjust the sidebar settings on the Sidebar tab.
          Learn more about configuring your content type
        </InformationSection>
      </Box>
    </Flex>
  );
};

type InformationSectionProps = {
  url: string;
  children: string;
  linkText: string;
  marginTop?: Spacing;
  marginBottom?: Spacing;
  dataTestId?: string;
};
function InformationSection(props: InformationSectionProps) {
  return (
    <Paragraph
      marginBottom={props.marginBottom ? props.marginBottom : 'spacing2Xs'}
      marginTop={props.marginTop ? props.marginTop : 'spacingXs'}>
      {props.children}{' '}
      <TextLink
        icon={<ExternalLinkIcon />}
        alignIcon="end"
        href={props.url}
        target="_blank"
        data-testid={props.dataTestId}
        rel="noopener noreferrer">
        {props.linkText}
      </TextLink>
      <Text> .</Text>
    </Paragraph>
  );
}

export default ConfigScreen;
