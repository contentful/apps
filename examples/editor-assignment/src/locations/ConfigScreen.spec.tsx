import React from 'react';
import ConfigScreen from './ConfigScreen';
import { render } from '@testing-library/react';
import { mockCma, mockSdk } from '../../test/mocks';

jest.mock('@contentful/react-apps-toolkit', () => ({
  useSDK: () => mockSdk,
  useCMA: () => mockCma,
}));

describe('Config Screen component', () => {
  it('Component text exists', async () => {
    const configScreen = render(<ConfigScreen />);

    // simulate the user clicking the install button
    await mockSdk.app.onConfigure.mock.calls[0][0]();

    expect(configScreen.getByText('Assign to Content Type Sidebar')).toBeInTheDocument();
    expect(configScreen.getByText('Assign to Content Type Field (Short Text)')).toBeInTheDocument();
  });
});
