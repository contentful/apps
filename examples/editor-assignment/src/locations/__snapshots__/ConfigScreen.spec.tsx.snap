// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config Screen component Component text exists 1`] = `
<div>
  <div
    class="css-tpeyg1"
  />
  <div
    class="css-jwwzlh"
  >
    <div
      class="css-1plp4qr"
    >
      <form
        data-test-id="cf-ui-form-label"
      >
        <h1
          class="css-cjax0m"
          data-test-id="cf-ui-heading"
        >
          Editor assignment example
        </h1>
        <ul
          class="css-1xin0wb"
          data-test-id="cf-ui-accordion"
        >
          <li
            class="css-bi3edf"
            data-test-id="cf-ui-accordion-item-id_787"
          >
            <h2
              class="css-9yunkd"
              data-test-id="cf-ui-accordion-header"
            >
              <button
                aria-controls="accordion-panel--id_787"
                aria-expanded="false"
                class="css-xsxndb"
                id="accordion--id_787"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-1xzxirq"
                  data-test-id="cf-ui-icon"
                  viewBox="0 0 9 18"
                >
                  <path
                    d="M1.1 5.9l3.4 3.4 3.4-3.4 1.1 1-4.5 4.5L0 6.9l1.1-1z"
                  />
                  <path
                    d="M0 0h9v18H0V0z"
                    fill="none"
                  />
                </svg>
                Assign to Content Type Sidebar
              </button>
            </h2>
            <div
              aria-hidden="true"
              aria-labelledby="accordion--id_787"
              class="css-18pon2m"
              data-test-id="cf-ui-accordion-panel"
              role="region"
            >
              <div
                class="css-1kmbcan"
              >
                <div
                  class="css-sjn9mf"
                >
                  <label
                    class="css-1qx9imn"
                    data-test-id="cf-ui-checkbox"
                    for="checkbox_1096"
                  >
                    <input
                      aria-checked="false"
                      class="css-gxjzre"
                      id="checkbox_1096"
                      role="checkbox"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="css-1wlznqq"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-1xugees"
                        data-test-id="cf-ui-icon"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                        />
                      </svg>
                    </span>
                    POST
                  </label>
                </div>
              </div>
            </div>
          </li>
          <li
            class="css-bi3edf"
            data-test-id="cf-ui-accordion-item-id_792"
          >
            <h2
              class="css-9yunkd"
              data-test-id="cf-ui-accordion-header"
            >
              <button
                aria-controls="accordion-panel--id_792"
                aria-expanded="false"
                class="css-xsxndb"
                id="accordion--id_792"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-1xzxirq"
                  data-test-id="cf-ui-icon"
                  viewBox="0 0 9 18"
                >
                  <path
                    d="M1.1 5.9l3.4 3.4 3.4-3.4 1.1 1-4.5 4.5L0 6.9l1.1-1z"
                  />
                  <path
                    d="M0 0h9v18H0V0z"
                    fill="none"
                  />
                </svg>
                Assign to Content Type Field (Short Text)
              </button>
            </h2>
            <div
              aria-hidden="true"
              aria-labelledby="accordion--id_792"
              class="css-18pon2m"
              data-test-id="cf-ui-accordion-panel"
              role="region"
            >
              <div
                class="css-1kmbcan"
              >
                
                <div
                  class="css-sjn9mf"
                >
                  <label
                    class="css-1qx9imn"
                    data-test-id="cf-ui-checkbox"
                    for="checkbox_9977"
                  >
                    <input
                      aria-checked="false"
                      class="css-gxjzre"
                      id="checkbox_9977"
                      role="checkbox"
                      type="checkbox"
                      value=""
                    />
                    <span
                      class="css-1wlznqq"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-1xugees"
                        data-test-id="cf-ui-icon"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20.7071 5.29289C21.0976 5.68342 21.0976 6.31658 20.7071 6.70711L9.70711 17.7071C9.51957 17.8946 9.26522 18 9 18C8.73478 18 8.48043 17.8946 8.29289 17.7071L3.29289 12.7071C2.90237 12.3166 2.90237 11.6834 3.29289 11.2929C3.68342 10.9024 4.31658 10.9024 4.70711 11.2929L9 15.5858L19.2929 5.29289C19.6834 4.90237 20.3166 4.90237 20.7071 5.29289Z"
                        />
                      </svg>
                    </span>
                    POST
                  </label>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </form>
    </div>
  </div>
</div>
`;
